File: Semana10py/Cajero.py
class Cajero:
    saldo = 1000

    def Menu(self) -> int:
        opc = -1
        while opc != 0:
            print("\nBienvenido al cajero.")
            print("[1] Consultar Saldo")
            print("[2] Depositar")
            print("[3] Retirar")
            print("[0] Salir")

            opc = int(input("Ingrese la opcion: "))

            match opc:
                case 1:
                    print(f"Su saldo es: {self.Consultar()}")
                case 2:
                    monto = float(input("Ingrese el monto a depositar: "))
                    self.Depositar(monto)
                case 3:
                    monto = float(input("Ingrese el monto a retirar: "))
                    self.Retirar(monto)
                case 0:
                    print("Gracias por usar el cajero.")
                case _:
                    print("Opcion invalida.")

        return opc

    def Consultar(self) -> float:
        return Cajero.saldo
    
    def Depositar(self, monto: float) -> None:
        while(True):
            if monto < 0:
                print("\nNo se puede depositar 0 o un valor negativo.")
                monto = float(input("Ingrese nuevamente el monto a depositar: "))

            elif monto == 0:
                print("Deposito cancelado.")
                break
            else:
                Cajero.saldo += monto
                print(f"Deposito exitoso. Saldo actual: {Cajero.saldo}")
                break

    def Retirar(self, monto: float) -> None:
        while(True):
            if monto < 0:
                print("\nNo se puede retirar 0 o un valor negativo.")
                monto = float(input("Ingrese nuevamente el monto a retirar:"))

            elif monto == 0:
                print("Retiro cancelado.")
                break

            elif monto > Cajero.saldo:
                print("Saldo insuficiente.")
                monto = float(input("Ingrese nuevamente el monto a retirar: "))

            else:
                Cajero.saldo -= monto
                print(f"Retiro exitoso. Saldo actual: {Cajero.saldo}")
                break

File: Semana10/Cajero.cs
using System;

namespace Semana10
{
    class Cajero
    {
        private static double saldo = 1000;
        public static int OperacionesBanco()
        {

            int opc;
            Console.WriteLine("Bienvenido al Cajero");
            do
            {
                Console.WriteLine("\n[1]Consultar Saldo");
                Console.WriteLine("[2] Depositar dinero");
                Console.WriteLine("[3] Retirar dinero ");
                Console.WriteLine("[0] Salir");
                Console.Write("\nElija una opcion: ");
                opc = Convert.ToInt32(Console.ReadLine());



                switch (opc)
                {
                    case 1:
                        Console.WriteLine($"Su saldo es: {Consultar()}");
                        break;

                    case 2:
                        Console.Write("Ingrese el monto a depositar (0 para cancelar): ");
                        double Deposito = Convert.ToDouble(Console.ReadLine());
                        Depositar(Deposito);
                        break;

                    case 3:
                        Console.Write("Ingrese el monto a retirar: ");
                        double retiro = Convert.ToDouble(Console.ReadLine());
                        Retirar(retiro);
                        break;

                    case 0:
                        Console.WriteLine("Gracias por usar el cajero.");
                        break;

                    default:
                        Console.WriteLine("Opcion invalida");
                        break;

                }
            } while (opc != 0);

            return opc;
        }
        public static double Consultar()
        {
            return saldo;
        }

        public static void Depositar(double monto)
        {
            while (true) 
            {
                if (monto < 0)
                {
                    Console.WriteLine("No se puede depositar un valor negativo.");
                    Console.Write("Ingrese el monto a depositar (0 para cancelar): ");
                    monto = Convert.ToDouble(Console.ReadLine());
                }
                else if (monto == 0)
                {
                    Console.WriteLine("Depósito cancelado.");
                    return;
                }
                else
                {
                    saldo += monto;
                    Console.WriteLine($"Depósito exitoso. Saldo actual: {saldo}");
                    return; 
                }

            } 
        }

        public static void Retirar (double monto)
        {
            do
            {
                if (monto < 0)
                {
                    Console.WriteLine("\nNo se puede retirar 0 o un valor negativo.");
                    Console.Write("Ingrese nuevamente el monto a retirar: ");
                    monto = Convert.ToDouble(Console.ReadLine());
                }
                else if (monto == 0)
                {
                    Console.WriteLine("Retiro Cancelado.");
                    return;
                }
                else if (monto > saldo)
                {
                    Console.WriteLine("\nNo se puede retirar un monto superior a tu saldo.");
                    Console.Write("Ingrese nuevamente el monto a retirar: ");
                    monto = Convert.ToDouble(Console.ReadLine());
                }
                else
                {
                    saldo -= monto;
                    Console.WriteLine($"Retiro exitoso. Saldo actual: {saldo}");
                    return;
                }


            } while (true);


        }
    }
}